code/game/machinery/omnibox.dm

```diff
@@ -0,0 +1,264 @@
+/obj/machinery/power/omnibox
+	name = "Omnibox"
+	desc = "A cutting edge device capable of automatically converting vast amounts of energy into unfathomably complex machinery."
+	icon = 'icons/obj/machines/omnibox.dmi'
+	icon_state = "dominator"
+	density = 1
+	anchored = FALSE
+	verb_say = "states"
+	var/drain_rate = 0	// amount of power to drain per tick
+	var/power_drained = 0 		// has drained this much power
+	var/active = FALSE
+	var/goal = 100
+	var/optional_message
+	var/list/possible_machines = list(
+		/datum/omnibox/factory
+	)
+
+/obj/machinery/power/omnibox/examine(mob/user)
+	..()
+	if(!active)
+		user << "<span class='notice'>The [src] seems to be inactive.</span>"
+	else
+		user << "<span class='notice'>The [src] is converting approximately [drain_rate] kilowatts of power every cycle (2 seconds). It has consumed [power_drained] kilowatts so far.</span>"
+
+/obj/machinery/power/omnibox/attackby(obj/item/O, mob/user, params)
+	if(!active)
+		if(istype(O, /obj/item/wrench))
+			if(!anchored && !isinspace())
+				connect_to_network()
+				user << "<span class='notice'>You secure the [src] to the floor.</span>"
+				anchored = TRUE
+			else if(anchored)
+				disconnect_from_network()
+				user << "<span class='notice'>You unsecure and disconnect the [src].</span>"
+				anchored = FALSE
+			playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
+			return
+	return ..()
+
+/obj/machinery/power/omnibox/attack_hand(mob/user)
+	..()
+	if (!anchored)
+		user << "<span class='warning'>This device must be anchored by a wrench!</span>"
+		return
+	interact(user)
+
+/obj/machinery/power/omnibox/attack_ai(mob/user)
+	interact(user)
+
+/obj/machinery/power/omnibox/attack_paw(mob/user)
+	interact(user)
+
+/obj/machinery/power/omnibox/interact(mob/user)
+	if (!anchored)
+		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
+		return
+	if(!Adjacent(user) && (!isAI(user)))
+		return
```

Comment:
57's if is not what you want it to do.

Reply:
I honestly just need a better template for these non-TGUI interfaces. Everyone seems to handle these differently and its hard to tell which interface code is good and which uses a bunch of outdated checks.

Reply:
This is actually correct, I was reading it as `||`